pulp_smash/tests/ostree/api_v2/test_sync_publish.py

```diff
@@ -0,0 +1,447 @@
+    """Create two OSTree repositories, with and without feed."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create two repositories."""
+        super(CreateTestCase, cls).setUpClass()
+        client = api.Client(cls.cfg, api.json_handler)
+        cls.bodies = tuple((_gen_ostree_repo_body() for _ in range(2)))
+        cls.bodies[1]['importer_config'] = {'feed': utils.uuid4()}
+        # RPM plugin doesn't validate url, puppet plugin does
+        cls.repos = [client.post(REPOSITORY_PATH, body) for body in cls.bodies]
+        cls.importers_iter = [
+            client.get(urljoin(repo['_href'], 'importers/'))
+            for repo in cls.repos
+        ]
+        for repo in cls.repos:
+            cls.resources.add(repo['_href'])  # mark for deletion
+
+    def test_id_notes(self):
+        """Validate the ``id`` and ``notes`` attributes for each repo."""
+        for key in ('id', 'notes'):
+            for body, attrs in zip(self.bodies, self.repos):
+                with self.subTest((key, body, attrs)):
+                    self.assertIn(key, attrs)
+                    self.assertEqual(body[key], attrs[key])
+
+    def test_number_importers(self):
+        """Each repository should have only one importer."""
+        for i, importers in enumerate(self.importers_iter):
+            with self.subTest(i=i):
+                self.assertEqual(len(importers), 1, importers)
+
+    def test_importer_type_id(self):
+        """Validate the ``importer_type_id`` attribute of each importer."""
+        key = 'importer_type_id'
+        for body, importers in zip(self.bodies, self.importers_iter):
+            with self.subTest((body, importers)):
+                self.assertIn(key, importers[0])
+                self.assertEqual(body[key], importers[0][key])
+
+    def test_importer_config(self):
+        """Validate the ``config`` attribute of each importer."""
+        key = 'config'
+        for body, importers in zip(self.bodies, self.importers_iter):
+            with self.subTest((body, importers)):
+                self.assertIn(key, importers[0])
+                self.assertEqual(body['importer_' + key], importers[0][key])
+
+
+class SyncValidFeedTestCase(_BaseTestCase):
+    """With a valid feed given, the sync completes with no reported errors."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create an OSTree repository with a valid feed and branch."""
+        super(SyncValidFeedTestCase, cls).setUpClass()
+        client = api.Client(cls.cfg, api.json_handler)
+        body = _gen_ostree_repo_body()
+        body['importer_config']['feed'] = _VALID_FEED
+        body['importer_config']['branches'] = [_VALID_BRANCHES[0]]
+        repo = client.post(REPOSITORY_PATH, body)
+        client.response_handler = api.echo_handler
+        cls.sync_report = client.post(
+            urljoin(repo['_href'], 'actions/sync/'),
+            {'override_config': {}}
+        )
+        cls.sync_report.raise_for_status()
```

Comment:
I don't see the logic behind setting the client's response handler to `api.echo_handler`, then calling `raise_for_status()` on each response. The following should be equivalent:

```python
client.response_handler = api.safe_handler
cls.sync_report = client.post(...)
```